eng/test-job.yml

```diff
@@ -0,0 +1,129 @@
+        inputs:
+          artifactName: ${{ format('{0}_{1}_{2}_build', parameters.osGroup, parameters.archType, parameters.buildConfig) }}
+          targetPath: $(Build.SourcesDirectory)/bin/Product/$(osGroup).$(archType).$(buildConfigUpper)
+    - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+      - task: DownloadPipelineArtifact@0
+        displayName: Download product build pipeline artifact
+        inputs:
+          artifactName: ${{ format('{0}_{1}_{2}_build', parameters.osGroup, parameters.archType, parameters.buildConfig) }}
+          targetPath: $(Build.SourcesDirectory)\bin\Product\Windows_NT.$(archType).$(buildConfigUpper)
+
+      
+    # Build tests
+    - ${{ if or(eq(parameters.osGroup, 'Linux'), eq(parameters.osGroup, 'OSX')) }}:
+        # TODO: enable crossgen in build-test.sh
+      - script: ./build-test.sh $(buildConfig) $(archType) $(priorityArg) $(crossgenArg)
+        displayName: Build pri-0 tests
+    - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+      - script: build-test.cmd $(buildConfig) $(archType) $(priorityArg) $(crossgenArg)
+        displayName: Build pri-0 tests
+
+
+    # Prepare tests for helix
+    - ${{ if or(eq(parameters.osGroup, 'Linux'), eq(parameters.osGroup, 'OSX')) }}:
+      - script: ./Tools/dotnetcli/dotnet msbuild tests/helixprep.proj /p:CORE_ROOT=$(Build.SourcesDirectory)/bin/tests/$(osGroup).$(archType).$(buildConfigUpper)/Tests/Core_Root /p:__BuildType=$(buildConfig) /p:__BuildArch=$(archType) /p:UsePython=true
+        # TODO: remove UsePython argument once we've removed generated wrappers in helixprep.proj
+        displayName: Prepare tests for helix
+    - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+        # TODO: remove UsePython argument once we've removed generated wrappers in helixprep.proj
+      - script: .\Tools\dotnetcli\dotnet.exe msbuild tests\helixprep.proj /p:CORE_ROOT=$(Build.SourcesDirectory)\bin\tests\Windows_NT.$(archType).$(buildConfigUpper)\tests\core_root /p:__BuildType=$(buildConfig) /p:__BuildArch=$(archType) /p:UsePython=true
+        displayName: Prepare tests for helix
+
+
+    # Send tests to helix
+    - ${{ if or(eq(parameters.osGroup, 'Linux'), eq(parameters.osGroup, 'OSX')) }}:
+      - script: ./Tools/dotnetcli/dotnet msbuild tests/helixpublishwitharcade.proj /t:Test $(scenariosArg)
+        displayName: Send test job to Helix
+        env:
+          ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+            # Access token variable for internal project
+            HelixAccessToken: $(HelixApiAccessToken)
+          ${{ if eq(variables['System.TeamProject'], 'public') }}:
+            # Access token variable for public project
+            HelixAccessToken: $(BotAccount-dotnet-github-anon-kaonashi-bot-helix-token)
+    - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+      - script: .\Tools\dotnetcli\dotnet msbuild tests\helixpublishwitharcade.proj /t:Test $(scenariosArg)
+        displayName: Send test job to Helix
```

Comment:
Out of pure curiosity: does this helix thing means that the build agent is busy doing nothing for an hour while the tests are executed on some other machine?

Reply:
For now, yes, but there's a flag that can be set to control whether it waits on the helix jobs. I think we ultimately want test reporting to be tied to the pipeline job completion, so we might need to find a way to have helix trigger a pipeline job to show test results. @chcosta, do you have any recommendations here?